internal/server/handlers/rbac/database_cluster_restore_test.go

```diff
@@ -0,0 +1,405 @@
+					"g, bob, role:test",
+				),
+				wantErr: ErrInsufficientPermissions,
+			},
+			{
+				desc: "missing read permission for database-cluster-credentials on 'cluster1'",
+				policy: newPolicy(
+					"p, role:test, database-cluster-restores, create, default/cluster1",
+					"p, role:test, database-cluster-backups, read, default/cluster1",
+					"p, role:test, database-cluster-restores, read, default/cluster1",
+					"g, bob, role:test",
+				),
+				wantErr: ErrInsufficientPermissions,
+			},
+			{
+				desc: "missing read permission for database-cluster-backups on 'cluster1'",
+				policy: newPolicy(
+					"p, role:test, database-cluster-restores, create, default/cluster1",
+					"p, role:test, database-cluster-credentials, read, default/cluster1",
+					"p, role:test, database-cluster-restores, read, default/cluster1",
+					"g, bob, role:test",
+				),
+				wantErr: ErrInsufficientPermissions,
+			},
+			{
+				desc: "missing read permission for database-cluster-restores on 'cluster1'",
+				policy: newPolicy(
+					"p, role:test, database-cluster-restores, create, default/cluster1",
+					"p, role:test, database-cluster-credentials, read, default/cluster1",
+					"p, role:test, database-cluster-backups, read, default/cluster1",
+					"g, bob, role:test",
+				),
+				wantErr: ErrInsufficientPermissions,
+			},
+		}
+
+		ctx := context.WithValue(context.Background(), common.UserCtxKey, "bob")
+		for _, tc := range testCases {
+			t.Run(tc.desc, func(t *testing.T) {
+				t.Parallel()
+				k8sMock := newConfigMapMock(tc.policy)
+				enf, err := rbac.NewEnforcer(ctx, k8sMock, zap.NewNop().Sugar())
+				require.NoError(t, err)
+				next := next()
+
+				h := &rbacHandler{
+					next:       next,
+					log:        zap.NewNop().Sugar(),
+					enforcer:   enf,
+					userGetter: testUserGetter,
+				}
+
+				_, err = h.CreateDatabaseClusterRestore(ctx, &everestv1alpha1.DatabaseClusterRestore{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "restore1",
+						Namespace: "default",
+					},
+					Spec: everestv1alpha1.DatabaseClusterRestoreSpec{
+						DBClusterName: "cluster1",
+					},
+				})
+				assert.ErrorIs(t, err, tc.wantErr)
+			})
+		}
+	})
+
+	t.Run("UpdateDatabaseClusterRestore", func(t *testing.T) {
+		next := func() *handlers.MockHandler {
+			h := &handlers.MockHandler{}
+			h.On("UpdateDatabaseClusterRestore", mock.Anything, mock.Anything).Return(&everestv1alpha1.DatabaseClusterRestore{}, nil)
+			return h
+		}
+
+		testCases := []struct {
+			desc    string
+			policy  string
+			wantErr error
+		}{
+			{
+				desc: "admin",
+				policy: newPolicy(
+					"g, bob, role:admin",
+				),
+			},
+			{
+				desc: "success",
+				policy: newPolicy(
+					"p, role:test, database-cluster-restores, update, default/cluster1",
+					"g, bob, role:test",
+				),
+			},
+			{
+				desc:    "missing update permission for database-cluster-restores on 'cluster1'",
+				policy:  newPolicy(),
+				wantErr: ErrInsufficientPermissions,
+			},
```

Comment:
Should the Update require similar permissions to Create? I mean these :point_down: 
```
					"p, role:test, database-cluster-credentials, read, default/cluster1",
					"p, role:test, database-cluster-backups, read, default/cluster1",
					"p, role:test, database-cluster-restores, read, default/cluster1",
```

Reply:
I'm actually not sure, but I did think about it. For now I just added the test cases based on the current RBAC implementation..

But I think it makes sense to have the same permissions and create..

I've noted that there are some minor RBAC permission changes required for DBB and DBR, and planned to do them in a separate PR. Shall I include this in that PR?

Reply:
Sure, feel free to handle this in a follow-up PR.